packages/shared/src/components/squads/layout/SquadDirectoryNavbar.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { ComponentProps, ReactElement } from 'react';
+import classNames from 'classnames';
+import {
+  Button,
+  ButtonProps,
+  ButtonSize,
+  ButtonVariant,
+} from '../../buttons/Button';
+import { useHorizontalScrollHeader } from '../../HorizontalScroll/useHorizontalScrollHeader';
+import { ArrowIcon } from '../../icons';
+import { useViewSize, ViewSize } from '../../../hooks';
+
+interface SquadNavbarItemProps extends Omit<ComponentProps<'li'>, 'onClick'> {
+  buttonSize: ButtonSize;
+  isActive: boolean;
+  label: string;
+  onClick: ButtonProps<'a'>['onClick'];
+  path?: string;
+}
+
+export const SquadDirectoryNavbarItem = ({
+  buttonSize,
+  className,
+  isActive,
+  label,
+  onClick,
+  path,
+  ...attrs
+}: SquadNavbarItemProps): ReactElement => (
+  <li
+    {...attrs}
+    className={classNames(
+      'relative py-3',
+      'after:absolute after:bottom-0 after:left-0 after:right-0 after:mx-auto after:w-14 after:border-b-2',
+      {
+        'after:hidden': !isActive,
+      },
+      className,
+    )}
+  >
+    <Button
+      aria-current={isActive ? 'page' : undefined}
+      aria-label={`Navigate to ${label}'s directory page`}
+      className="capitalize"
+      href={path}
+      onClick={onClick}
+      size={buttonSize}
+      tag={path ? 'a' : 'button'}
+      variant={isActive ? ButtonVariant.Float : ButtonVariant.Tertiary}
+    >
+      {label}
+    </Button>
+  </li>
+);
+
+export const SquadDirectoryNavbar = (
+  props: ComponentProps<'nav'>,
+): ReactElement => {
+  const { className, children, ...attrs } = props;
+  const {
+    ref,
+    onClickPrevious,
+    onClickNext,
+    isAtEnd,
+    isAtStart,
+    isOverflowing,
+  } = useHorizontalScrollHeader<HTMLUListElement>({
+    title: 'Squad directory navigation',
+  });
+  const isLaptop = useViewSize(ViewSize.Laptop);
+
+  return (
+    <nav
+      aria-label="Squad directory navigation"
+      className={classNames(
+        'relative -mx-4 border-b border-border-subtlest-tertiary px-4 laptop:border-0',
+        'flex-row flex-nowrap gap-2 laptop:flex',
+        className,
+      )}
+      {...attrs}
+    >
+      {!isAtStart && isOverflowing && isLaptop && (
```

Comment:
More wondering should the Horizontal nav not handle the buttons already?

Reply:
The HorizontalScroll component can't be adapted to this use case because its controls are in a different row from the content (due to the layout). I’ve abstracted the controls (to be used headless) from the hook and used them there.